docs/RFCS/error_handling.md

```diff
@@ -0,0 +1,97 @@
+presentation of errors.
+
+# Background
+
+Much of our code existing uses the ubiquitous `if err != nil { return err }`
+pattern, sometimes with the addition of context via an `errors.Wrapf` call.
+These errors can bubble up to our SQL connection handler or to a callsite that
+simply passes them to `panic`. Many callsites to `panic` or `log.Fatal` construct
+a string or error object containing both a statement of what went wrong as well
+the value(s) in question, e.g. `missing value in column %s` or
+`unexpected value: raw_bytes:"..."`.
+
+# Motivation
+
+Bubbling the original error back up, adding as much context as possible along
+the way, makes it easier to see what went wrong and thus figure out how to fix
+it. Unfortunately it also means that when our connection handler (pgwire) gets
+an error, or our panic handler recovers a crash, we can make very limited
+assumptions about what is actually in the error, and thus whether it is suitable
+for presentation to the user, inclusion in a diagnostic crash report, etc.
+
+Presenting users (SQL clients) with the details of some internal (e.g. KV) error
+is poor UX: we should strive to provide useful and actionable errors that the
+user can do something about. Internal errors should be logged internally (to
+our log files) where the administrator can find and troubleshoot then.
+
+Error messages that contain potentially sensitive user data cannot be included
+in crash reports, and at some point in the future, if/when we add support for
+encrypting stores, we may want an option to avoid inclusion in (unencrypted) log
+files as well.
+
+# Detailed Design
+
+## Presentation to Users (i.e. errors to SQL clients)
+TL;DR: sql executor will only render and return `pgerror.Error`-derived errors.
+
+While implementing SQL execution, we must consciously choose when constructing
+an error whether it is intended for presentation to a user, and use `pgerror`
+helpers when it is to correctly indicate that, along with its matching error
+code.
+
+Errors *not* intended for user consumption should be wrapped in some sort of
+`InternalError` which, when presented to the user, simply renders something like
+`internal error (logged as <nodeid:id>)` while logged the wrapped error to its
+logs with the unique ID, such that a DBA with log access, presented with the
+message shown to the user, could then find and trouble it.
+
+Initially any non-`pgerror` error arriving at the executor can simply be logged
+and then an `internal error` returned.
+
+If we wanted to leverage the compiler to help us ensure we were returning the
+correct error types before they got to the executor, we could change the
+signature of `planNode` to return a `pgerror`.
+
+ See #5452.
+
+## Crash Reporting (and potentially Non-sensitive Log Files)
+
+### Tag Values with `log.Safe{}` Wrapper
+Add `type Safe struct {v interface{}}` for tagging args as non-sensitive.
+
+If a value wrapped in `Safe` is recovered by a panic handler, it is known to be
```

Comment:
eh, -1 on `Share` -- that's one use case for the info captured, but as discussed above, this is also intended to flag which fields should{ not} be redacted if/when we want to allow writing non-sensitive log files. This is a tag on a value, as to whether or not it is known to be free of sensitive info.

To add to this bike shed, you'll likely want to repeat this multiple times on a line to log multiple safe values, so we want something short and concise `log.Nonsensitive{}`  or `log.Reportable{}`would be kinda gross.

example usage might looks like 
```
if !valid(c.Arg) {
 return log.Errorf("unexpected value processing %s: %v", log.Safe{c.Type}, c.Arg)
}
```